test/abort.js

```diff
@@ -48,6 +48,7 @@ test('abort request if server responds early', function (t) {
         res.end()
         clearTimeout(timer)
         server.close()
+        client.destroy() // Destroy keep-alive agent.
```

Comment:
Reviewer note: Every other change below (all to files under test/...), with the exception of a change in test/lib/utils.js, is about calling `client.destroy()` to properly close active handles (timers, the keep-alive net.Socket) so that the test completes with having to wait for long timeouts.